tests/test_structures.py

```diff
@@ -0,0 +1,73 @@
+import ase
+import numpy as np
+import pytest
+
+
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_structure_manager_widget(structure_data_object):
+    """Test the `StructureManagerWidget`."""
+    import aiidalab_widgets_base as awb
+
+    structure_manager_widget = awb.StructureManagerWidget(
+        importers=[
+            awb.StructureUploadWidget(title="From computer"),
+        ],
+        editors=[
+            awb.BasicStructureEditor(title="Basic Editor"),
+        ],
+    )
+    structure_manager_widget.input_structure = structure_data_object
+
+    assert structure_manager_widget.structure is not None
+    assert isinstance(structure_manager_widget.structure, ase.Atoms)
+
+    # Store structure and check that it is stored.
+    structure_manager_widget.store_structure()
+    assert structure_manager_widget.structure_node.is_stored
+    assert structure_manager_widget.structure_node.pk is not None
+
+    # Simulate the structure modification.
+    new_structure = structure_manager_widget.structure.copy()
+    new_structure[0].position += [0, 0, 1]
+
+    structure_manager_widget.structure = new_structure
+    assert structure_manager_widget.structure_node.pk is None
```

Comment:
I don't understand why this will be None?

Reply:
Because the structure is not the saved one anymore, since it was modified. But I am not sure it is guaranteed to have `pk==None` for an unsaved structure. Probably, a better option would be to do the following:
```suggestion
    assert not structure_manager_widget.structure_node.is_stored
```

Reply:
I see. I think the change you proposed may not perfect. If I understand correctly if the `new_structure` is stored, the `structure_node` will become the stored node? Such as:

```python
    structure_manager_widget.structure = new_structure
    new_structure.store()
    assert structure_manager_widget.structure_node.is_stored is True
``` 

I think checking pk is fine, please just add a comment on it.

Reply:
Ah, you already did the change. Then it is fine, I have no preference, the comment is also not very necessary I think.